src/gl_fns.rs

```diff
@@ -129,6 +129,47 @@ impl Gl for GlFns {
         }
     }
 
+    fn read_pixels_into_buffer_with_stride(
```

Comment:
this seems like a higher level function, especially considering that it needs to save and restore some state...

Reply:
What do you mean by a higher level function? Something that shouldn't be in gleam?

Reply:
Yeah. It's supposed to be an API wrapper. Once it starts changing and restoring the state internally, it becomes higher level, which is inconsistent with the rest of the API. Is there any other method that tries to restore state? Cursory look doesn't reveal any.

In your PR, what was the blocker of having this method in WR side? You have access to `read_pixels_into_buffer`, and you have `pixel_store_i` already. I suppose the only blocker was an assert in `calculate_length`. So instead of adding a method, I think the right way to do here is making the assert aware of the current row length (i.e. assert doing `self.ffi_gl_.GetIntegerv(ffi::PACK_ROW_LENGTH, &mut row_length as _);`).